src/pages/Curation/components/Accordion/index.tsx

```diff
@@ -828,62 +823,64 @@ const Accordion = () => {
         </div>
       </div>
     );
-  }, [emotionsCheckedValue]);
-
-  const handleResetValues = useCallback(() => {
-    setPersonCheckedValue('');
-    setAge('');
-    setJobCheckedValue('');
-    setPurposeCheckedValue('');
-    setRelationshipCheckedValue('');
-    setCheckedList([]);
-    setEmotionsCheckedValue('');
-  }, []);
+  }, [emotions]);
 
   //TODO : POST API (선물 추천받기)
   const getRecommendedPresents = async (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
     e.preventDefault();
 
-    if (
-      personCheckedValue === '' ||
-      purposeCheckedValue === '' ||
-      age === '' ||
-      relationshipCheckedValue === ''
-    ) {
-      alert('필수 항목들을 선택해 주세요.');
-      return;
-    }
-    setLoader(true);
     try {
       //넘길 값
-      const optionalValuesObj = {
-        person: optionalPersonCheckedValue,
-        age: optionalAge,
-        jobs: optionalJobCheckedValue,
-      };
+      setLoader(true);
       const valuesObj = {
-        person: personCheckedValue,
+        gender: personCheckedValue,
         age: age,
-        jobs: jobCheckedValue,
+        job: jobCheckedValue,
         purpose: purposeCheckedValue,
-        relationship: relationshipCheckedValue,
-        price: prices,
-        productType: checkedList,
+        relation: relationCheckedValue,
+        minPrice: prices[0],
+        maxPrice: prices[1],
+        category: checkedList.join(','),
         emotion: emotionsCheckedValue,
       };
 
-      //res.json이 200일 때 처리로 변경 필요.
-      //현재는 단순 로직 보여주기 위해 타임아웃을 걸었음
+      //total checked count
+      const arr = [];
+      arr.push(personCheckedValue);
+      arr.push(age);
+      arr.push(jobCheckedValue);
+      arr.push(purposeCheckedValue);
+      arr.push(relationCheckedValue);
+      arr.push(prices[0]);
+      checkedList.forEach((item: any) => {
+        arr.push(item);
+      });
+      arr.push(emotionsCheckedValue);
+      const result = arr.filter(e => e);
+
+      //API 요청
+      const data = await recommendPresents(valuesObj);
+
+      //매칭률 계산
+      data.forEach((item: any, idx: number) => {
+        const percentage = Math.floor(((item.score + 2) / result.length) * 100);
+        data[idx].percentage = percentage;
+      });
+
+      //매칭률 별 정렬
+      data.sort((a: any, b: any) => {
+        return b.percentage - a.percentage;
+      });
+
       setTimeout(() => {
         setLoader(false);
-        console.log(optionalValuesObj, valuesObj);
-        location.href = '/curation/present/recommendation';
+        navigate('/curation/present/recommendation', {
+          state: [data, data[0].percentage, result, valuesObj],
```

Comment:
어떤 값들인지 명시해서 넘겨주면 조금 더 가독성이 좋아질 것 같습니다.
```suggestion
          const A = data;
          const pricePercentage = data[0]percentage; // 예시
          const C = result;
          const D = valuesObj;
          
          // ...
          
          state: [A, pricePercentage, C, D],
```